### ISSUE TITLE
OBS under Linux with Pipewire records audio with lots of stutter from Samson G-Track Pro USB mic

### ISSUE THREAD

**Issue by PaddyMac** [obsproject/obs-studio#8980]
### Operating System Info

Other

### Other OS

Gentoo

### OBS Studio Version

29.1.1

### OBS Studio Log URL

https://obsproject.com/logs/ZLYOpkHfTEOJRBXn

### Expected Behavior

I expected audio to be recorded with clarity.

### Current Behavior

Audio from my EVGA XR1 Lite capture card is recorded fine. But audio recorded from my Samson G-Track Pro USB microphone has major stuttering under Linux while using Pipewire. It was fine when I was using Pulseaudio. I dual-boot with Windows 10, and everything works fine with OBS under Windows with the same hardware.

### Steps to Reproduce

1. Use Pipewire for audio under Linux
2. Record audio with a Samson G-Track Pro USB mic
3. Realize during playback of recorded/streamed audio that there is a lot of stutter.
...

### Anything else we should know?

I don't know if this is a Pipewire issue or an OBS issue. I spent some time on the Pipewire IRC troubleshooting, and I can get some pretty good quality audio with almost no stuttering if I force a sample rate of 44100 and a quant of 2048. But, at the same time, other applications, such as Audacity, will record audio with zero stuttering without need of any modifications to anything. The only tested application experiencing problems is OBS. I also tried recording audio from the built-in mic on my USB webcam, and there is no stutter from it, either. If I had to guess, there might be an issue because the Samson G-Track Pro records audio in a 24-bit format. There is a message when OBS is run from the console which says "pulse-input: Sample format s24le not supported by OBS,using float32le instead for recording". So maybe that at least is a part of the problem.

**Comment by kkartaltepe** [comment]
> It was fine when I was using Pulseaudio

We only support pulseaudio, so if pipewire's pulseaudio backend behaves differently then this should be considered a pipewire bug. As such I will close this out.

**Comment by PaddyMac** [comment]
After your response, I decided to revert my system to using Pulseaudio, and I no longer have any problems with stuttering audio.

**Comment by PaddyMac** [comment]
Just a further comment in case anyone else comes across this bug report, I decided to try Pipewire as my audio system again. I upgraded to 0.3.74. There was still stuttering. I then upgraded my Linux kernel from 6.1.38 to 6.4.3. I'm pretty sure I'd also tried 6.2.x and 6.3.x kernels at some point. But, despite a lack of extensive testing, it appears this may have been a Linux kernel bug. I now have perfectly fine audio via my microphone in OBS with no stuttering, lag, or other noticeable problem.
